src/servers/grpc_server.cc

```diff
@@ -1729,20 +1722,15 @@ SharedMemoryControlHandler::Process(Handler::State* state, bool rpc_ok)
   SharedMemoryControlResponse& response = state->response_;
 
   if (state->step_ == START) {
-    TRTSERVER_SharedMemoryBlock* smb = nullptr;
-
     TRTSERVER_Error* err = nullptr;
     if (request.has_register_()) {
       if (request.register_().has_system_shared_memory()) {
         // system shared memory
-        err = smb_manager_->CpuCreate(
-            &smb, request.register_().name(),
+        RETURN_IF_ERR(shm_manager_->AddSystemSharedMemory(
+            request.register_().name(),
             request.register_().system_shared_memory().shared_memory_key(),
             request.register_().system_shared_memory().offset(),
-            request.register_().byte_size());
-        if (err == nullptr) {
-          err = TRTSERVER_ServerRegisterSharedMemory(trtserver_.get(), smb);
-        }
+            request.register_().byte_size()));
```

Comment:
Just notice that why there is underscore for `register_()`?

Reply:
It seems to be a protoc behavior. I am not sure why does it adds an under-score after the field name.

Reply:
register is probably a keyword or reserved. It therefore adds '_' to the end of the variable name